src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -46,9 +46,13 @@ public partial struct SyntaxValueProvider
         var syntaxHelper = _context.SyntaxHelper;
 
         // Create a provider that provides (and updates) the global aliases for any particular file when it is edited.
-        var individualFileGlobalAliasesProvider = this.CreateSyntaxProvider(
-            static (n, _) => n is ICompilationUnitSyntax,
-            static (context, _) => getGlobalAliasesInCompilationUnit(context.SyntaxHelper, context.Node)).WithTrackingName("individualFileGlobalAliases_ForAttribute");
+        //var individualFileGlobalAliasesProvider = this.CreateSyntaxOnlyProvider(
+        //    static (n, _) => n is ICompilationUnitSyntax).Select(
+        //    (n, _) => getGlobalAliasesInCompilationUnit(syntaxHelper, n)).WithTrackingName("individualFileGlobalAliases_ForAttribute");
+
+        var syntaxTreesProvider = _context.CompilationProvider.SelectMany((c, _) => c.SyntaxTrees);
+        var individualFileGlobalAliasesProvider = syntaxTreesProvider.Select(
+            (s, c) => getGlobalAliasesInCompilationUnit(syntaxHelper, s.GetRoot(c))).WithTrackingName("individualFileGlobalAliases_ForAttribute");
```

Comment:
@jaredpar @chsienki it's unclear to me if we'd actually need the 'pure syntax provdier' given that we can just simulate it by getting the syntax-trees from teh compilation-provider.  these are already pure-syntax, and already support the equality semantics we need here.

Reply:
Nice.